### DISCUSSION TITLE
Support for range queries

### DISCUSSION THREAD

**Discussion by vtripolitakis** [quickwit-oss/quickwit#817] [Q&A] [answered]
Hello and congrats for your great work.

As per https://github.com/quickwit-inc/quickwit/issues/140 I see that for the time being, range queries are not supported. Is any plan on the roadmap to add such support? 

Thanks in advance.

**Comment by fulmicoton** [comment, marked as answer]
Hello @vtripolitakis ! Thanks for your question.

We already support timestamp range queries. If this is your use case it should work out of the box.

Timestamps have a special status in Quickwit.
We  does split pruning based on the time range you sent.
Our merge strategy also tries to merge splits that are contiguous in time, in order to keep this split pruning as powerful as possible.

If you need range query on another field, it is on our roadmap. You can expect to see in the 2nd quarter of next year I would say.


We will add range query support relatively soon but it will most likely

**Comment by fmassot** [comment]
@vtripolitakis range queries will be supported in the next release 0.6 planned for the end of the month.
